src/nvim/globals.h

```diff
@@ -194,12 +194,13 @@ EXTERN int compl_cont_status INIT(= 0);
 EXTERN int cmdmsg_rl INIT(= false);  // cmdline is drawn right to left
 EXTERN int msg_col;
 EXTERN int msg_row;
-EXTERN int msg_scrolled;        /* Number of screen lines that windows have
-                                * scrolled because of printing messages. */
-EXTERN int msg_scrolled_ign INIT(= FALSE);
-/* when TRUE don't set need_wait_return in
-   msg_puts_attr() when msg_scrolled is
-   non-zero */
+EXTERN int msg_scrolled;        // Number of screen lines that windows have
+                                // scrolled because of printing messages.
+EXTERN bool msg_scrolled_ign INIT(= false);
+EXTERN bool msg_did_scroll INIT(= false);
+
+// when TRUE don't set need_wait_return in msg_puts_attr()
+// when msg_scrolled is non-zero
```

Comment:
this should go above `msg_scrolled_ign`.

Reply:
Thanks, but is #9530 change.